### PR TITLE
Add toggle-able labels in SiteNav components

### PR DESCRIPTION
**Pull request by slafleche**
I had done most of https://github.com/vanilla/knowledge/issues/617, but I was missing this last piece to make the guide categories toggable instead of links in the side panel nav.

Other updates
- Converted button loader to use CSS in JS styles to have the colour changeable.
- Renamed buttonVariables const for clariy
- Added links variables in globalVariables
- Fixed max-width of search in header

### CODE REVIEW

File: library/src/scripts/components/forms/select/SearchBar.tsx [hunk 1]
```diff
@@ -119,7 +121,7 @@ export default class SearchBar extends React.Component<IProps, IState> {
                 menuIsOpen={this.isMenuVisible}
                 classNamePrefix={this.prefix}
                 className={classNames(this.prefix, className)}
-                placeholder={this.props.placeholder}
+                placeholder={this.props.placeholder || ""}
```
Comment (initvector) [inline library/src/scripts/components/forms/select/SearchBar.tsx]: It looks like you're using `||` to implement a default value for this property. Would it be better to set `placeholder` to an empty string in `defaultProps`?

File: library/src/scss/components/_siteNav.scss [hunk 2]
```diff
@@ -146,8 +145,11 @@ $siteNav-node_padding               : $utility-baseUnit - $siteNav-node_borderWi
 
 
 .siteNavNode-buttonOffset {
+    position: relative;
     display: flex;
     justify-content: flex-end;
     width: $siteNav-nodeToggle_width;
     margin-left: -#{$siteNav-nodeToggle_width};
+    top: #{(37 / 2)}px;
```
Comment (alex-mtl) [inline library/src/scss/components/_siteNav.scss]: Seem like hardcoded value (37). Could we base it on some global variable value? do we have default `line-height` or `siteNav-button-height` or `siteNav-height` ?

Comment (slafleche) [reply library/src/scss/components/_siteNav.scss]: I'll fix it when I convert the styles to JS.

Comment (alex-mtl) [reply library/src/scss/components/_siteNav.scss]: Why not now? Never mind

**Pull request merged**